lib/storage/search.go

```diff
@@ -182,6 +182,9 @@ func (s *Search) reset() {
 //
 // Init returns the upper bound on the number of found time series.
 func (s *Search) Init(qt *querytracer.Tracer, storage *Storage, tfss []*TagFilters, tr TimeRange, maxMetrics int, deadline uint64) int {
+	so := getSearchOptions(deadline, "search")
```

Comment:
I think, it's kinda a blocker for merging this feature. It's needed to distinguish somehow `/query_range` `/query` and other search request calls. But it's not possible without changing protocol between vmstorage and vmselect.

Reply:
We've discussed this and decided to keep this one as it is right now in order to avoid breacking changes in RPC protocol comptatibility. We might reconsider this when RPC protocol will be easier to extend https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7054